### PR TITLE
Added avro library

### PR DESCRIPTION
**Pull request by zirrock**
Adds avro library submodule and a test using it

### CODE REVIEW

File: test/boost/avro_test.cc [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2016 ScyllaDB
```
Comment (haaawk) [inline test/boost/avro_test.cc:2]: s/2016/2020

File: test/boost/avro_test.hh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#ifndef AVRO_TEST_HH
```
Comment (haaawk) [inline test/boost/avro_test.hh:1]: Could we merge this file into avro_test.cc or is it used somewhere else as well?

File: test/boost/avro_test.cc (hunk 1 above)
Comment (zirrock) [reply test/boost/avro_test.cc]: Fixed

File: test/boost/avro_test.hh (hunk 2 above)
Comment (zirrock) [reply test/boost/avro_test.hh]: It would be hard to merge, because I'm declaring a template instance there

**Comment by zirrock** [comment]
Squashed and added missing file

**Comment by haaawk** [comment]
I think you need to modify test.py as well to make sure avro test is run with all other unit tests.

**Comment by haaawk** [comment]
> > I think you need to modify test.py as well to make sure avro test is run with all other unit tests.
> 
> Not necessary anymore; now test.py has an auto-detection [pattern](https://github.com/scylladb/scylla/blob/master/test.py#L181) for each test type.

Cool. I will check whether the test is run automatically. Thanks @dekimir. How does the build file look like, in your opinion?

**Pull request merged**